src/ChannelServer/World/Entities/Components/Buff.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.Data.Database;
+
+namespace Melia.Channel.World
+{
+	/// <summary>
+	/// Modeled after Skill class
+	/// </summary>
+	public class Buff
+	{
+		/// <summary>
+		/// The buff's owner.
+		/// </summary>
+		public ICombatEntity Caster { get; }
+
+		/// <summary>
+		/// The buff's target.
+		/// </summary>
+		public ICombatEntity Target { get; }
+
+		/// <summary>
+		/// Returns the buff's id.
+		/// </summary>
+		public BuffId Id { get; }
+
+		/// <summary>
+		/// Returns the buff's associated skill id.
+		/// Default value is 1.
+		/// </summary>
+		public int SkillId { get; }
+
+		/// <summary>
+		/// The buff's data from the buff database.
+		/// </summary>
+		public BuffData Data { get; }
+		
+		/// <summary>
+		/// The buff's duration, separate from database duration
+		/// because database duration is inaccurate.
+		/// </summary>
+		public int Duration { get; }
+
+		/// <summary>
+		/// Index in world collection?
+		/// </summary>
+		public int Handle { get; private set; }
+
+		/// <summary>
+		/// Returns the buff's overbuff count. If this value reaches the buff's
+		/// maximum overbuff, the buff has additional effects based on this
+		/// counter.
+		/// </summary>
+		public int OverbuffCounter { get; private set; }
+
+		/// <summary>
+		/// The buff's removal time
+		/// </summary>
+		public DateTime RemovalTime { get; set; }
+
+		/// <summary>
+		/// If the buff has an update time
+		/// </summary>
+		/// <returns></returns>
+		public bool HasUpdateTime() => this.Data.UpdateTime != 0;
```

Comment:
This is supposed to be a property, right? In that case you can remove the parantheses.